### ISSUE TITLE
False Alert on unused dependency

### ISSUE THREAD

**Issue by darlenya** [depcheck/depcheck#737]
### Bug Description

I have a dependency which is alerted as unused. 
The reason for it is the folder Structure like this:
root
   |- someDir
       |- lib
           - fileUsingTheDependency

The dependencies used in the files in the 'lib' folder are ignored.
After moving the file to a folder called 'lub' the dependency was not alerted any more.

### Versions

node: v16.14.2
depcheck 1.4.3

**Comment by martinsagat** [comment]
@darlenya have you added parsers?

`
const options = {
  parsers: {
      // the target parsers
      '**/*.js': depcheck.parser.es6,
      '**/*.jsx': depcheck.parser.jsx,
      '*.ts': depcheck.parser.typescript,
      '*.vue': depcheck.parser.vue,
  },
 }

depcheck(__dirname, options).then((unused) => {
  /* eslint-disable */
  console.log('Unused dependencies: ', unused.dependencies) // an array containing the unused dependencies // eslint-disable-line no-use-before-define
  console.log('Unused devDependecies: ', unused.devDependencies) // an array containing the unused devDependencies
  console.log('Dependencies missing in package.json: ', unused.missing) // a lookup containing the dependencies missing in `package.json` and where they are used
  console.log(unused.using) // a lookup indicating each dependency is used by which files
  console.log(unused.invalidFiles) // files that cannot access or parse
  console.log(unused.invalidDirs) // directories that cannot access
  /* eslint-disable */
})
`

**Comment by cope** [comment]
@martinsagat how does one add parsers in .depcheckrc file?

**Comment by martinsagat** [comment]
@cope you don't add them to file, you can pass them in parameter:

--parsers="*.js:es6,*.jsx:jsx,*.json:json1&json2" etc..

**Comment by cope** [comment]
I think this may work:

```
parsers:
  '**/*.js': ['es6']
  '**/*.jsx': ['jsx']
  '**/*.ts': ['typescript']
  '**/*.vue': ['vue']
```

you gotta be able to add them to a file?

See: https://github.com/depcheck/depcheck/blob/main/doc/pluggable-design.md#use-parser-from-a-configuration-file

**Comment by martinsagat** [comment]
@cope you are right actually, glad you found what you were looking for 🙂

**Comment by darlenya** [comment]
Hi, no I have not added any parsers. It is just that the files in a folder Called 'lib' are ignored. 
I do not understand why this parser setting is the solution. Could you explain it?

**Comment by cope** [comment]
@darlenya no, I'm just guessing since I had a similar problem but on my side there is no lib folder...

**Comment by darlenya** [comment]
> @darlenya no, I'm just guessing since I had a similar problem but on my side there is no lib folder...

ok, thanks. Was a little bit confused :-)
